### ISSUE TITLE
Can't use ON DELETE CASCADE

### ISSUE THREAD

**Issue by Azramis** [msawczyn/EFDesigner#76]
Hi,

I've tried to generate bi-directionnal table relation on one-to-many, with the "ON DELETE CASCADE", but it seems that it generated nothing to do that.
I had to put the code by myself.
```csharp
.OnDelete(DeleteBehavior.Cascade);
```` 

Maybe I'm doing something wrong ? Eventually an bug ?

**Comment by dcastenholz** [comment]
I think that if you change the generation code for UnidirectionalAssociation to:

`
  if (association.SourceDeleteAction != DeleteAction.Default)
  {
    string willCascadeOnDelete = (association.SourceDeleteAction == DeleteAction.Cascade).ToString().ToLowerInvariant();
    segments.Add($"WillCascadeOnDelete({willCascadeOnDelete})");
  }
`

it might fix it.

**Comment by msawczyn** [comment]
Nice. I'll put that on the list of things to hit for the next release. I'm working the `IdentityDbContext` right now and am pretty focused on that, but from the investigation you've done it appears to be a quick fix.

**Comment by Azramis** [comment]
Thanks guys for response.
@dcastenholz  My problem is on BidirectionnalAssociation, not sure that your fix apply to me.

In my case I change the generation code for Bidirectionnal into this:

```csharp
if ((/*association.SourceDeleteAction != DeleteAction.Default &&*/ association.SourceRole == EndpointRole.Principal) ||
             (/*association.TargetDeleteAction != DeleteAction.Default &&*/ association.TargetRole == EndpointRole.Principal))
{
		 
    DeleteAction deleteAction = (association.SourceRole == EndpointRole.Principal)
                                          ? association.SourceDeleteAction
                                          : association.TargetDeleteAction;
    switch (deleteAction)
    {
         case DeleteAction.None:
            segments.Add("OnDelete(DeleteBehavior.Restrict)");
            break;
         case DeleteAction.Cascade:
            segments.Add("OnDelete(DeleteBehavior.Cascade)");
            break;
	 default:
            segments.Add("OnDelete(DeleteBehavior.SetNull)");
            break;
   }
}
```

It generate the ``.OnDelete(DeleteBehavior.Cascade);``.

You can see that I manage default case manually, in order to fix a bug when using SQLite with EFCore. Default value for ``OnDelete`` in EFCore is ``ClientSetNull``, but this value seems not be applicated on SQLite (RESTRICT is applicated...), so I set the "different but same" SetNull.

Maybe my fix is not good, but it works for me.

**Comment by msawczyn** [comment]
Fixed in v1.3.0.4 - thanks!

**Comment by msawczyn** [comment]
Closing since fix was released.
